### ISSUE TITLE
DynamicArray treats uninitialized objects as initialized

### ISSUE THREAD

**Issue by CyberShadow** [dlang-community/containers#109]
Test case:

```d
import std.stdio;

import containers.dynamicarray;

struct S
{
	bool initialized;
	@nogc:
	@disable this();
	this(int) { initialized = true; }
	~this() { assert(initialized); }
}

void main()
{
	auto s = S(0);

	DynamicArray!S arr;
	arr.insertBack(s);
}
```

The assert fails because `insertBack` uses assignment to overwrite uninitialized memory:

```d
		arr = cast(typeof(arr)) allocator.allocate(T.sizeof * 4);
		...
		arr[l++] = value;
```

Instead, something like [emplace](https://dlang.org/phobos/std_conv.html#emplace) must be used.

**Comment by Panke** [comment]
What's the state of this?

**Comment by CyberShadow** [comment]
It was fixed when #110 was merged.
